python/lsst/ip/diffim/detectAndMeasure.py

```diff
@@ -267,9 +267,9 @@ def run(self, science, matchedTemplate, difference, selectSources,
             difference image.
         difference : `lsst.afw.image.ExposureF`
             Result of subtracting template from the science image.
-        selectSources : `lsst.afw.table.SourceCatalog`, optional
+        selectSources : `lsst.afw.table.SourceCatalog`
             Identified sources on the science exposure.
-        idFactory : `lsst.afw.table.IdFactory`
+        idFactory : `lsst.afw.table.IdFactory`, optional
```

Comment:
This documentation fix correctly matches the implementation, but that implementation is wrong. `selectSources` should be removed entirely, and `idFactory` should not have a default. Go ahead with these cleanups, and I will fix the code on DM-36043.